cardano-api/src/Cardano/Api/IO.hs

```diff
@@ -99,15 +117,13 @@ handleFileForWritingWithOwnerPermission path f = do
     (targetDir, targetFile) = splitFileName path
 #endif
 
-newtype OutputFile = OutputFile
-  { unOutputFile :: FilePath
-  }
-  deriving Generic
-  deriving newtype (Eq, Ord, Show, IsString, ToJSON, FromJSON)
-
-writeByteStringFile :: MonadIO m => FilePath -> ByteString -> m (Either (FileError ()) ())
+writeByteStringFile :: ()
+  => MonadIO m
+  => File content Out
+  -> ByteString
+  -> m (Either (FileError ()) ())
 writeByteStringFile fp bs = runExceptT $
-  handleIOExceptT (FileIOError fp) $ BS.writeFile fp bs
+  handleIOExceptT (FileIOError (unFile fp)) $ BS.writeFile (unFile fp) bs
```

Comment:
`unFile` is an escape hatch to get the `FilePath` within.  We use it to implement file operations like `writeByteStringFile`.